GliaWidgets/Sources/CallVisualizer/VideoCall/View/VideoCallView.swift

```diff
@@ -0,0 +1,399 @@
+            if currentOrientation.isLandscape {
+                hideLandscapeBarsAfterDelay()
+            } else {
+                headerTopConstraint.constant = 0
+                buttonBarBottomConstraint.constant = 0
+            }
+        }
+    }
+}
+
+// MARK: - Props
+
+extension CallVisualizer.VideoCallView {
+    struct Props: Equatable {
+        let style: CallStyle
+        let callDuration: String?
+        let connectViewProps: CallVisualizer.VideoCallView.ConnectView.Props
+        let buttonBarProps: CallVisualizer.VideoCallView.CallButtonBar.Props
+        let headerTitle: String
+        let operatorName: String?
+        let remoteVideoStream: CoreSdkClient.StreamView?
+        let localVideoStream: CoreSdkClient.StreamView?
+        let topLabelHidden: Bool
+        let endScreenShareButtonHidden: Bool
+        let connectViewHidden: Bool
+        let topStackAlpha: CGFloat
+        let headerProps: Header.Props
+    }
+}
+
+// MARK: - Private
+
+private extension CallVisualizer.VideoCallView {
+    func renderProps() {
+        renderCallDuration = props.callDuration
+        connectView.props = props.connectViewProps
+        buttonBar.props = props.buttonBarProps
+        renderOperatorName = props.operatorName
+        renderRemoteVideoStream = props.remoteVideoStream
+        renderLocalVideoStream = props.localVideoStream
+        topLabel.isHidden = props.topLabelHidden
+        header.endScreenShareButton.isHidden = props.endScreenShareButtonHidden
+        connectView.isHidden = props.connectViewHidden
+        topStackView.alpha = props.topStackAlpha
+        header.props = props.headerProps
+    }
+
+    func adjustVideoViews() {
+        adjustRemoteVideoView()
+    }
+
+    func adjustRemoteVideoView() {
+        remoteVideoViewHeightConstraint.constant = frame.size.height
+        remoteVideoViewWidthConstraint.constant = frame.size.width
+    }
+
+    func showBars(duration: TimeInterval) {
+        layoutIfNeeded()
+        UIView.animate(withDuration: duration) {
+            self.headerTopConstraint.constant = 0
+            self.buttonBarBottomConstraint.constant = 0
+            self.layoutIfNeeded()
+        }
+    }
+
+    func hideBars(duration: TimeInterval) {
+        layoutIfNeeded()
+        let newHeaderConstraint = -header.frame.size.height + safeAreaInsets.top
+        UIView.animate(withDuration: duration) {
+            self.headerTopConstraint.constant = newHeaderConstraint
+            self.buttonBarBottomConstraint.constant = self.buttonBar.frame.size.height
+            self.layoutIfNeeded()
+        }
+    }
+
+    func hideLandscapeBars() {
+        guard currentOrientation.isLandscape else { return }
+        hideBars(duration: 0.3)
+    }
+
+    func hideLandscapeBarsAfterDelay() {
+        hideBarsWorkItem?.cancel()
+        let hideBarsWorkItem = DispatchWorkItem { self.hideLandscapeBars() }
+        self.hideBarsWorkItem = hideBarsWorkItem
+        DispatchQueue.main.asyncAfter(
```

Comment:
DispatchQueue should not be used directly, but injected via Environment.GCD. Otherwise it will impact snapshot testing.

Reply:
How would injecting it work since Environment needs to be Equatable?

Reply:
Yes

Reply:
fixed [879ee57](https://github.com/salemove/ios-sdk-widgets/pull/494/commits/879ee575c7ed5ee486ed0552d0ae122d3a4c24cd)